### PR TITLE
Automate API docs. Add docstrings.

### PR DESCRIPTION
**Pull request by khrapovs**
## Issue

- API documentation is incomplete and not automated.

## Changes

- Automate API docs.
- Add docstrings.

## Change severity

- [ ] Major (breaking change)
- [ ] Minor (new feature)
- [X] Patch (improvement, bug fix)

### CODE REVIEW

File: docs/gen_ref_pages.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+nav = mkdocs_gen_files.Nav()
+
+for path in sorted(Path(SRC).rglob("*.py")):
+    module_path = path.relative_to(SRC).with_suffix("")
+    doc_path = path.relative_to(SRC).with_suffix(".md")
+    full_doc_path = Path("reference", doc_path)
+
+    parts = tuple(module_path.parts)
+
+    if parts[-1] == "__init__":
```
Comment (shannendorothee) [inline docs/gen_ref_pages.py:21]: what only in this condition process should continue?

File: docs/gen_ref_pages.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        continue
+    if parts[-1] == "__main__":
```
Comment (shannendorothee) [inline docs/gen_ref_pages.py:23]: same here

File: docs/gen_ref_pages.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        continue
```
Comment (shannendorothee) [inline docs/gen_ref_pages.py:24]: else, what happens if this case is not fulfilled?

File: docs/gen_ref_pages.py (hunk 1 above)
Comment (khrapovs) [reply docs/gen_ref_pages.py:21]: `continue` means "go to next loop iteration". effectively skip anything that matches the `if` condition. try commenting this out and see what happens to the docs...

File: docs/gen_ref_pages.py (hunk 3 above)
Comment (khrapovs) [reply docs/gen_ref_pages.py:24]: this is just a simple filter for docstrings to parse

File: mkdocs.yaml [hunk 4]
```diff
@@ -1,12 +1,44 @@
...
         python:
           options:
             docstring_style: numpy
+            show_source: false
+
+markdown_extensions:
+  - pymdownx.arithmatex:
+      generic: true
+
+extra_javascript:
```
Comment (akankshatoshniwal) [inline mkdocs.yaml:42]: what does this field do?

Comment (khrapovs) [reply mkdocs.yaml:42]: try commenting it out and see http://127.0.0.1:8000/reference/order_matching/order_book/. LaTeX rendering

File: docs/gen_ref_pages.py (hunk 1 above)
Comment (shannendorothee) [reply docs/gen_ref_pages.py:21]: understood

**Review by shannendorothee — APPROVED** [review]

**Review by akankshatoshniwal — APPROVED** [review]

**Pull request merged**